### PR TITLE
comby 0.10.0 (new formula)

### PR DESCRIPTION
**Pull request by rvantonder**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/comby.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class Comby < Formula
+  desc "Tool for changing code across many languages"
+  homepage "https://comby.dev"
+  url "https://github.com/comby-tools/comby/archive/0.10.0.tar.gz"
+  sha256 "081633c100ca6696760e21bd3decd01666c9ad4946b370dc90412e55a324435e"
+
+  depends_on "gmp" => :build
+  depends_on "opam" => :build
+  depends_on :x11 => :build
```
[the remark this answers is not in the document]
Comment (rvantonder) [reply Formula/comby.rb]: Hi @SMillerDev. I'd love to figure out a way, and was surprised when [the build failed](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/50664/version=catalina/testReport/junit/brew-test-bot/catalina/install___build_bottle_comby/) on an X11-related dependency.

```
# cc -c -O2 -fno-strict-aliasing -fwrapv -Wall  -I/usr/X11/include -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE  -I../../runtime -o open.o open.c 
# In file included from open.c:19:
# ./libgraph.h:17:10: fatal error: 'X11/Xlib.h' file not found
# #include <X11/Xlib.h>
#          ^~~~~~~~~~~~
```

This appears to be coming from trying to compile the OCaml compiler, and [a formula already exists for that](https://github.com/Homebrew/homebrew-core/blob/master/Formula/ocaml.rb) without any X11 issues, so I'm not sure what's going on (maybe the include path is wrong?). A cursory search for this issue didn't bring up an obvious solution. Subsequent builds did get past this error when I added the `:x11` dependency, but that doesn't seem ideal.

[the remark this answers is not in the document]
Comment (rvantonder) [reply Formula/comby.rb]: @SMillerDev good news! I tracked down the OCaml dependency on X11 and it is indeed a default dependency in 4.08.1. The new good news is that it is [moved out in 4.09.0](https://sourcegraph.com/github.com/ocaml/ocaml@4.10/-/blob/Changes#L485), so I will target a more recent version of the compiler. 

I also came across the discussion in #41066 about this issue and how you're discouraging x11. So, helpful not If this crops up elsewhere: the solution is to either target `4.09.0`, or set the compiler configure flag [-no-graph](https://sourcegraph.com/github.com/ocaml/ocaml@0933593596d6ebe1b99e8344b9ceb0c181aad6b6/-/blob/INSTALL.adoc#L128-129) for packages targetting `4.08.1` (and possibly earlier).

**Comment by rvantonder** [comment]
@SMillerDev the changes I proposed in the thread works and the `X11` dependency is gone :tada: Please take another look :)

**Comment by SMillerDev** [comment]
I'm a bit weirded out by it creating an exe but that's nothing you can change. Looks good. Thanks for contributing to homebrew @rvantonder without people like you it would be impossible to maintain homebrew to the level people have come to expect of us.

**Comment by rvantonder** [comment]
The `.exe` is a design choice by the [build system](https://dune.build/) and I agree it does scan weird--it's meant to denote 'executable' generally, without with the windows connotation.

Thanks for the quick turnaround and, of course, for homebrew!

**Review by chenrui333 — APPROVED** [review]
👍

**Comment by chenrui333** [comment]
Thanks @rvantonder for your contribution to Homebrew! :tada: 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**